tools/cli/include/nearobject/cli/NearObjectCliHandler.hxx

```diff
@@ -99,8 +100,9 @@ struct NearObjectCliHandler
 
 private:
     NearObjectCli* m_parent;
+    std::shared_ptr<::uwb::UwbDevice> m_activeDevice;
     std::shared_ptr<::uwb::UwbSession> m_activeSession;
-    std::shared_ptr<NearObjectCliUwbSessionEventCallbacks> m_sessionEventCallbacks;
+    std::shared_ptr<::uwb::UwbSessionEventCallbacks> m_sessionEventCallbacks;

```

Comment:
Why don't we want to use NearObjectCliUwbSessionEventCallbacks here?

Reply:
I'm guessing its because NearObjectCliUwbSessionEventCallbacks is the derived class and doesn't need to be part of the public interface

Reply:
`NearObjectCliUwbSessionEventCallbacks` is just an implementation of the interface that's actually required, `::uwb::UwbSessionEventCallbacks`. How it implements it (and the additional stuff it exposes, if any) is just an implementation detail that `NearObjectCliHandler.cxx` should know the minimum required about. Typically, storing an instance to the least-derived class in a hierarchy is safest since it reduces, and in many cases, eliminates the possible interactions with an instance, which means it's easier to understand and the domain of test outputs is minimized. In this case, `NearObjectClieHandler` only needs to know how to construct the instance with the specific bits it needs (the session ended callback), so there's no need to store a more specific derived instance.